### PR TITLE
test(ivy): migrate styling tests from PR #34000

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR contains acceptance tests migrated from the PR #34000 (https://github.com/mhevery/angular/commits/pr_34000).

I did a number of modifications to the migrated tests. For the ones from https://github.com/mhevery/angular/commit/3555c1a198bb5a379fc5e12fe93fdef3fa3c510a:
* `should allow multiple styling bindings to work alongside property/attribute bindings` was migrated with slight modifications (removed bindings not used in asserts);
* `should allow host styling on the root element with external styling` was migrated as-is;
* `should apply camelCased classes` - migrated as `should apply camelCased class names` 
* `should apply camel-cased [style] properties` - migrated as `should convert camelCased style property names to snake-case`;
* `should recover from an error thrown in styling bindings` - was migrated with modifications (tighter asserts);
* `should apply template-level style bindings by writing to the style attribute directly` - there were 3 (!) tests with the same exact name and I migrated them as follows:

1. not migrated as was a low-level test asserting on the order of DOM manipulation (changed with Misko's implementation);
2. fixed (had a broken assert) and migrated as `should combine host class.foo bindings from multiple directives`;
3. migrated as `should combine static host classes with component "class" host attribute` 

https://github.com/mhevery/angular/commit/c77557534a272247172d545561f86cb37fa0ebd9	

Migrated 2 tests as `should allow a single style host binding on an element` and `should override class bindings when a directive extends another directive`

Some of the migrated tests are failing currently (ivy and / or VE) - more details in the in-line comments. So far I wasn't able to determine if a test or code is broken. Also some of the failures in VE are suspicious (looks like in practice there is a breaking change but I'm not sure we should have it....).

### CODE REVIEW

File: packages/core/test/acceptance/styling_spec.ts [hunk 1]
```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
     expect(getComputedStyle(div).width).toBe('10px');
   });
 
-  it('should allow classes with trailing and leading spaces in [ngClass]', () => {
+  it('should allow multiple styling bindings to work alongside property/attribute bindings', () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts:2627]: This test fails in VE with `Invalid CSS style: unsafe` (but works in ivy) which tells me that there is a difference wrt to sanitisation.

File: packages/core/test/acceptance/styling_spec.ts [hunk 2]
```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
...
+    });
+    const fixture = TestBed.createComponent(MyComp);
+    fixture.detectChanges();
+
+    const classList = (fixture.nativeElement.querySelector('div') as HTMLDivElement).classList;
+    expect(classList.contains('fooBar')).toBeTruthy();
+    expect(classList.contains('barFoo')).toBeTruthy();
+  });
+
+  it('should convert camelCased style property names to snake-case', () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts:2698]: This test fails in VE which tells me that VE wasn't making this conversion. Is is a deliberate breaking change?

File: packages/core/test/acceptance/styling_spec.ts [hunk 3]
```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
...
+    expect(() => fixture.detectChanges()).toThrow();
+
+    raiseWidthError = false;
+    fixture.detectChanges();
+    const div = fixture.nativeElement.querySelector('div') as HTMLDivElement;
+    expect(div.style.getPropertyValue('width')).toEqual('100px');
+    expect(div.style.getPropertyValue('height')).toEqual('200px');
+  });
+
+  it('should prioritize host bindings for templates first, then directives and finally components',
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts:2743]: This test fails in VE with `Invalid CSS style: unsafe` (but works in ivy) which tells me that there is a difference wrt to sanitisation.

File: packages/core/test/acceptance/styling_spec.ts [hunk 4]
```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
...
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('1 2 3');
+
+    root.classList.add('0');
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3');
+
+    // TODO(pk): why ???
+    fixture.componentInstance.classes = '1 2 3 4';
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3 4');
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/styling_spec.ts:2679]: This assert fails in VE with `Error: Expected '1 2 3 4' to equal '0 1 2 3 4'.`. Is is a deliberate breaking change?

File: packages/core/test/acceptance/styling_spec.ts (hunk 3 above)
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/styling_spec.ts:2743]: This test also fails with ivy with asserts on `elm.style.width`

**Comment by mhevery** [comment]
merged into https://github.com/angular/angular/pull/34616

**Pull request closed without merging**